### PR TITLE
nixos/magnetico: fixes

### PR DESCRIPTION
**Pull request by rnhmjoj**
###### Motivation for this change
Fix some stupid mistakes I did.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via `nixos/tests/magnetico.nix`
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/tests/magnetico.nix [hunk 1]
```diff
@@ -21,8 +22,8 @@ import ./make-test.nix ({ pkgs, ...} : {
       startAll;
       $machine->waitForUnit("magneticod");
       $machine->waitForUnit("magneticow");
-      $machine->succeed("${pkgs.curl}/bin/curl -u user:password http://localhost:8080");
-      $machine->succeed("${pkgs.curl}/bin/curl -u user:wrongpwd http://localhost:8080") =~ "Unauthorised." or die;
+      $machine->succeed("${pkgs.curl}/bin/curl -u user:password http://localhost:8081");
```
Comment (prusnak) [inline nixos/tests/magnetico.nix]: This could be `web.port` variable for consistency.

Comment (rnhmjoj) [reply nixos/tests/magnetico.nix]: Ok, done.

**Comment by rnhmjoj** [comment]
ping

**Comment by c0bw3b** [comment]
@GrahamcOfBorg test magnetico

**Comment by c0bw3b** [comment]
It's not referenced in all-tests.nix ?

**Comment by rnhmjoj** [comment]
> It's not referenced in all-tests.nix ?

Found another mistake.

**Comment by rnhmjoj** [comment]
I added the test to `all-tests.nix`, ported it to the python driver and fixed a race condition in.

**Comment by c0bw3b** [comment]
@GrahamcOfBorg test magnetico

**Review by c0bw3b — APPROVED** [review]

**Comment by rnhmjoj** [comment]
Thank you

**Pull request merged**